File: frontend/src/types/result.ts
export interface RankedPlan {
  rank: number;
  plan_id: number;
  plan_code: string;
  company_name: string;
  plan_name: string;
  annual_total: number;
  monthly_avg: number;
  saving_vs_standard: number;
  recommendation_reason: string;
  is_unsuitable: boolean;
}

export interface SubmitResponse {
  session_id: string;
  ranking: RankedPlan[];
}

File: backend/app/models/result.py
from datetime import datetime
from sqlalchemy import Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from app.database import Base


class CalculationResult(Base):
    __tablename__ = "calculation_results"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    session_id: Mapped[str] = mapped_column(
        String, ForeignKey("sessions.session_id", ondelete="CASCADE"), nullable=False
    )
    plan_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("plan_master.id", ondelete="CASCADE"), nullable=False
    )
    annual_total: Mapped[float] = mapped_column(Float, nullable=False)
    monthly_avg: Mapped[float] = mapped_column(Float, nullable=False)
    saving_vs_standard: Mapped[float] = mapped_column(Float, nullable=False)
    recommendation_reason: Mapped[str] = mapped_column(String, nullable=False)
    is_unsuitable: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    calculated_at: Mapped[datetime] = mapped_column(DateTime, nullable=False)
